src/components/TimeFrameDropdown/TimeFrameDropdown.js

```diff
@@ -1,19 +1,19 @@
-import React from 'react'
+import React, { useMemo } from 'react'
 import PropTypes from 'prop-types'
+import _map from 'lodash/map'
+import { useTranslation } from 'react-i18next'
 
 import Dropdown from '../../ui/Dropdown'
 import timeFrames from '../../util/time_frames'
 
 const TimeFrameDropdown = ({ tf, onChange }) => {
-  const options = []
-  const tfLabels = Object.keys(timeFrames)
-
-  for (let i = 0; i < tfLabels.length; ++i) {
-    options.push({
-      label: tfLabels[i],
-      value: timeFrames[tfLabels[i]],
-    })
-  }
+  const { t } = useTranslation()
+  const options = useMemo(() => {
+    return _map(timeFrames, (time) => ({
+      value: time,
+      label: t(`time.${time}`),
+    }))
+  }, [])
```

Comment:
shouldn't we add `t` here as a dependency, so the options will be updated when the language changes?
```suggestion
  }, [t])
```